tests/test_module_loader.py

```diff
@@ -318,3 +332,90 @@ def test_load_module_recursive_complex_module_dependencies(local_module_package_
         "complex_module_dependencies_mod2::submod",
     }
     assert loaded_namespaces == expected_namespaces
+
+
+def test_load_import_based_v2_project(local_module_package_index: str, snippetcompiler_clean) -> None:
+    """
+    A project needs to explicitly list its v2 dependencies in order to be able to load them. Import-based loading is not
+    allowed.
+    """
+    module_name: str = "minimalv2module"
+
+    def load(requires: Optional[List[Requirement]] = None) -> None:
+        project: Project = snippetcompiler_clean.setup_for_snippet(
+            f"import {module_name}",
+            autostd=False,
+            install_project=False,
+            python_package_sources=[local_module_package_index],
+            # make sure that even listing the requirement in project.yml does not suffice
+            project_requires=[InmantaModuleRequirement.parse(module_name)],
+            python_requires=requires,
+        )
+        project.load_module_recursive(install=True)
+
+    with pytest.raises(ModuleLoadingException, match=f"Failed to load module {module_name}"):
+        load()
+    # assert that it doesn't raise an error with explicit requirements set
+    load([Requirement.parse(ModuleV2Source.get_python_package_name(module_name))])
+
+
+@pytest.mark.parametrize("v1", [True, False])
+@pytest.mark.parametrize("explicit_dependency", [True, False])
+def test_load_import_based_v2_module(
+    local_module_package_index: str,
+    snippetcompiler_clean,
+    modules_dir: str,
+    modules_v2_dir: str,
+    tmpdir: py.path.local,
+    v1: bool,
+    explicit_dependency: bool,
+) -> None:
+    """
+    A module needs to explicitly list its v2 dependencies in order to be able to load them. Import-based loading is not
+    allowed.
+    """
+    main_module_name: str = "mymodule"
+    dependency_module_name: str = "minimalv2module"
+    index: PipIndex = PipIndex(os.path.join(str(tmpdir), ".my-index"))
+    libs_dir: str = os.path.join(str(tmpdir), "libs")
+    os.makedirs(libs_dir)
+
+    if v1:
+        shutil.copytree(os.path.join(modules_dir, "minimalv1module"), os.path.join(libs_dir, main_module_name))
```

Comment:
Why patch up a module to something which is actually completely different (I think)? 
This is not easy to read (I need to investigate both `minimalv1module` and all this code) and it doesn't very clearly convey what it does.

Is it not easier to use e.g. the `makemodule` helper?

Reply:
- The idea behind `minimalvxmodule` is that they are minimal so you shouldn't necessarily need to look at them to understand the code.
- I thought it wouldn't be worthwhile to create such a helper for v1 but maybe I was wrong about that one, I'll add it.

Reply:
done